bluepy/btle.py

```diff
@@ -500,12 +502,13 @@ def _getIncludedServices(self, startHnd=1, endHnd=0xFFFF):
         self._writeCmd("incl %X %X\n" % (startHnd, endHnd))
         return self._getResp('find')
 
-    def getCharacteristics(self, startHnd=1, endHnd=0xFFFF, uuid=None):
+    def getCharacteristics(self, startHnd=1, endHnd=0xFFFF, uuid=None,
+                           timeout=5):
         cmd = 'char %X %X' % (startHnd, endHnd)
         if uuid:
             cmd += ' %s' % UUID(uuid)
         self._writeCmd(cmd + "\n")
-        rsp = self._getResp('find')
+        rsp = self._getResp('find', timeout=timeout)
```

Comment:
this is cool however if the timeout is reached, it returns a None here and so then the code can't be gracefully handled.. maybe a raised error or more handling like what you did for readCharacteristic?